agent/app/agent.go

```diff
@@ -447,10 +447,15 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 	// Register the container instance
 	err = agent.registerContainerInstance(client, vpcSubnetAttributes)
 	if err != nil {
-		if isTransient(err) {
+		if isTerminal(err) {
+			// On unrecoverable error codes, agent should terminally exit.
+			seelog.Warnf("Agent will terminally exit, unable to register container instance: %v", err)
+			return exitcodes.ExitTerminal
+		} else if isTransient(err) {
 			return exitcodes.ExitError
 		}
-		return exitcodes.ExitTerminal
+		// Other errors are considered recoverable and will be retried.
+		return exitcodes.ExitError
```

Comment:
Why do we want to change the default?

Reply:
+1

Reply:
Pasting from convo: A terminal exit for the agent is a last resort case because it's painful in terms of task experience - so we want to be very specific about when to terminally exit rather than have it be the catch-all default.